vault/logical_system.go

```diff
@@ -6168,9 +6168,8 @@ func sanitizePath(path string) string {
 		path += "/"
 	}
 
-	for strings.HasPrefix(path, "/") {
-		path = path[1:]
-	}
+	// Remove all leading forward slashes and backslashes
+	path = strings.TrimLeft(path, "/\\")
```

Comment:
This might not be the correct based on my reading of the finding, i.e.
`has a leading slash, but not that it does not have '/' or '\' in its second position.`

For example, I think this would modify `\string` to `string`, but that it shouldn't. Based on my reading, I think this should only trim `/`, `//`, or `/\`.

Reply:
I might be misunderstanding - the way I understood this was that we not only have to check for a leading "/" but also a "//" and "/\". Is this what you're saying as well Violet?

Reply:
My reading is that it needs to check for `/`, `//`, and `/\` -- it shouldn't modify a path that starts with `\`

Reply:
Violet's comment is the way I understand it too

Reply:
For clarity here, the behavior we want is:

//my_string ==> my_string
/my_string ==> my_string
/\my_string ==> my_string
\/my_string ==> \/my_string
\\\\\//my_string ==> \\\\\//my_string

Reply:
Pushed a commit to address the concerns, please do let me know if it looks better now, many thanks!